Parsers/ASimAuditEvent/Parsers/vimAuditEventBarracudaWAF.yaml

```diff
@@ -0,0 +1,307 @@
+      | extend
+          EventSchema = "AuditEvent",
+          EventSchemaVersion = "0.1.0",
+          EventVendor = "Barracuda",
+          EventProduct = "WAF",
+          EventCount = toint(1)
+      | extend
+          Dvc = DeviceName, 
+          Operation = ProcessName,
+          DvcIpAddr = DeviceAddress,
+          NewValue = DeviceCustomString1,
+          SrcIpAddr = SourceIP,
+          EventMessage = Message,
+          OldValue = DeviceCustomString2,
+          DvcHostname = DeviceName,
+          ActorUsername = DestinationUserName,
+          Object = FileName,
+          ThreatConfidence = toint(ThreatConfidence),
+          EventStartTime = iff(isnotempty(FlexNumber2), unixtime_milliseconds_todatetime(tolong(ReceiptTime)-tolong(FlexNumber2)), unixtime_milliseconds_todatetime(tolong(ReceiptTime)))
+      | extend
+          Src = SrcIpAddr,
+          ActorUsernameType = iff(isnotempty(ActorUsername),"Simple",""),
+          ActorUserType = iff(isnotempty(ActorUsername), "Admin", ""),
+          User = ActorUsername,
+          Value = NewValue,
+          EventEndTime = EventStartTime
+      | extend
+          IpAddr = SrcIpAddr,
+          ValueType = iff(isnotempty(Value),"Other","")
+      | project-away
+          EventType_lookup,
+          ThreatConfidence,
+          CommunicationDirection,
+          AdditionalExtensions,
+          Device*,
+          Source*,
+          Destination*,
+          Activity,
+          LogSeverity,
+          ApplicationProtocol,
+          ProcessID,
+          ExtID,
+          Protocol,
+          Reason,
+          ReceiptTime,
+          SimplifiedDeviceAction,
+          OriginalLogSeverity,
+          ProcessName,
+          EndTime,
+          ExternalID,
+          File*,
+          ReceivedBytes,
+          Message,
+          Old*,
+          EventOutcome,
+          Request*,
+          StartTime,
+          Field*,
+          Flex*,
+          Remote*,
+          Malicious*,
+          severity,
+          ThreatSeverity,
+          IndicatorThreatType,
+          ThreatDescription,
+          _ResourceId,
+          SentBytes,
+          ReportReferenceLink,
+          Computer,
+          TenantId
+  };
+  union isfuzzy = true 
+      BarracudaCustom(
+          disabled=disabled,
+          starttime=starttime,
+          endtime=endtime,
+          srcipaddr_has_any_prefix=srcipaddr_has_any_prefix,
+          eventtype_in=eventtype_in,
+          eventresult=eventresult,
+          newvalue_has_any=newvalue_has_any,
+          operation_has_any=operation_has_any
+      ),
+      BarracudaCEF(
+          disabled=disabled,
+          starttime=starttime,
+          endtime=endtime,
+          srcipaddr_has_any_prefix=srcipaddr_has_any_prefix,
+          eventtype_in=eventtype_in,
+          eventresult=eventresult,
+          newvalue_has_any=newvalue_has_any,
+          operation_has_any=operation_has_any
+      )
```

Comment:
@jayeshprajapaticrest Why we added filtering to these functions separately? We should do union under a parent function and add filter to that parent function. Please refer this : https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

This is applicable to all Barracuda PRs. Please update

Reply:
applicable to Authentication, Network and Web PR

Reply:
Done in all Barracuda PRs. Authentication, Network and Web PR